compiler/rustc_middle/src/mir/coverage.rs

```diff
@@ -51,6 +51,23 @@ rustc_index::newtype_index! {
     pub struct ExpressionId {}
 }
 
+rustc_index::newtype_index! {
+    /// ID of a mcdc condition. Used by llvm to check mcdc coverage.
+    ///
+    /// Note that LLVM handles condition IDs as `int16_t`, so there is no need
+    /// to use a larger representation on the Rust side.
+    #[derive(HashStable)]
+    #[encodable]
+    #[orderable]
+    #[max = 0xFFFF]
```

Comment:
Note for future work: This ID probably doesn't need a max of 0xffff. (The other coverage IDs probably don't need 0xffff either.)

But trying to figure out the correct value right now is probably a waste of time, so we can leave it like this, and worry about changing it in a future PR.